sdk/core/azure-core/inc/azure/core/http/websockets/websockets.hpp

```diff
@@ -0,0 +1,369 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Azure Core APIs implementing the WebSocket protocol [RFC 6455]
+ * (https://www.rfc-editor.org/rfc/rfc6455.html).
+ */
+
+#pragma once
+
+#include "azure/core/context.hpp"
+#include "azure/core/http/http.hpp"
+#include "azure/core/http/transport.hpp"
+#include "azure/core/internal/client_options.hpp"
+#include <string>
+#include <vector>
+
+namespace Azure { namespace Core { namespace Http { namespace WebSockets {
+  namespace _detail {
+    class WebSocketImplementation;
+  }
+  namespace _internal {
+
+    enum class WebSocketFrameType : int
+    {
+      Unknown,
+      TextFrameReceived,
+      BinaryFrameReceived,
+      PeerClosedReceived,
+    };
+
+    enum class WebSocketErrorCode : uint16_t
```

Comment:
if the previous one is Type, shouldn't this also be WebSocketErrorCodeType

Reply:
WebSocketErrorCode is not the type of a websocket error code. It's the value of a WebSocket error.

The convention in the C++ SDK is that things that describe the kind of thing are called XxxType. But enumerations like error codes are not a "kind of thing" (unless you're getting excessively pedantic).